src/DynamoCoreWpf/Controls/DNAAutocompleteBar.xaml.cs

```diff
@@ -0,0 +1,170 @@
+using Dynamo.Graph.Nodes;
+using Dynamo.Graph.Workspaces;
+using Dynamo.Logging;
+using Dynamo.Models;
+using Dynamo.ViewModels;
+using System;
+using System.Linq;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Input;
+using System.Windows.Threading;
+
+namespace Dynamo.UI.Controls
+{
+    /// <summary>
+    /// Interaction logic for AutoCompleteSearchControl.xaml
+    /// Notice this control shares a lot of logic with InCanvasSearchControl for now
+    /// But they will diverge eventually because of UI improvements to auto complete.
+    /// </summary>
+    public partial class DNAAutocompleteBar : IDisposable
+    {
+        ListBoxItem HighlightedItem;
+
+        double currentX;
+
+        ListBoxItem currentListBoxItem;
+
+        /// <summary>
+        /// Node AutoComplete Search ViewModel DataContext
+        /// </summary>
+        private NodeAutoCompleteSearchViewModel ViewModel => DataContext as NodeAutoCompleteSearchViewModel;
+
+        public DNAAutocompleteBar(Window window, NodeAutoCompleteSearchViewModel viewModel)
+        {
+            this.Owner = window;
+            DataContext = viewModel;
+            InitializeComponent();
+            if (string.IsNullOrEmpty(DynamoModel.HostAnalyticsInfo.HostName) && Application.Current != null)
+            {
+                Application.Current.Deactivated += CurrentApplicationDeactivated;
+                if (Application.Current?.MainWindow != null)
+                {
+                    Application.Current.MainWindow.Closing += NodeAutoCompleteSearchControl_Unloaded;
+                }
+            }
+            HomeWorkspaceModel.WorkspaceClosed += this.CloseAutoCompletion;
+            ViewModel.IsOpen = true;
+            LoadAndPopulate();
+        }
+
+        private void NodeAutoCompleteSearchControl_Unloaded(object sender, System.ComponentModel.CancelEventArgs e)
+        {
+            if (string.IsNullOrEmpty(DynamoModel.HostAnalyticsInfo.HostName) && Application.Current != null)
+            {
+                Application.Current.Deactivated -= CurrentApplicationDeactivated;
+                if (Application.Current?.MainWindow != null)
+                {
+                    Application.Current.MainWindow.Closing -= NodeAutoCompleteSearchControl_Unloaded;
+                }
+            }
+            HomeWorkspaceModel.WorkspaceClosed -= this.CloseAutoCompletion;
+        }
+
+        private void CurrentApplicationDeactivated(object sender, EventArgs e)
+        {
+            CloseAutoCompletion();
+        }
+
+        private void LoadAndPopulate()
+        {
+            Analytics.TrackEvent(
+            Dynamo.Logging.Actions.Open,
+            Dynamo.Logging.Categories.NodeAutoCompleteOperations);
```

Comment:
We may need a new category for identifying cluster autocomplete specifically, which we can do later